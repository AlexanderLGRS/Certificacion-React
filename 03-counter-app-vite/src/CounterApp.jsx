import PropTypes from 'prop-types'
import { useState } from 'react';
// import "./styles.css"
export default function CounterApp({ value, interval }) {
    const [newValue, setNewValue] = useState(value)

    const incrementValue = () => {
        setNewValue(newValue + interval)
    }
    const decrementValue = () => {
        setNewValue(newValue - interval)
    }
    const resetValue = () => {
        setNewValue(value)
    }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{newValue}</h2>
            <button aria-label='button-increment' onClick={incrementValue}> + </button>
            <button aria-label='button-decrement' onClick={decrementValue}> - </button>
            <button aria-label='button-reset' onClick={resetValue}> Reset </button>
        </>

    )
}
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}