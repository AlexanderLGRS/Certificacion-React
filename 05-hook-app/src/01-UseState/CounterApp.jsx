import { useState } from "react"

export const CounterApp = () => {
    const [values, setValues] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30
    })
    const { counter1, counter2, counter3 } = values
    return (
        <>
            <h1>Counter: {counter1}</h1>
            <h1>Counter: {counter2}</h1>
            <h1>Counter: {counter3}</h1>
            <hr />
            <button className="btn btn-success" onClick={() => setValues({
                ...values,
                counter1: counter1 + 1
            })}
            >+1</button>
        </>
    )
}