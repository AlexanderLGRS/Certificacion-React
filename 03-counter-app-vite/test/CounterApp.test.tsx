import { fireEvent, getByRole, render, screen } from '@testing-library/react';
import React from 'react';
import CounterApp from '../src/CounterApp';

describe('Pruebas en App', () => {
  const initialValue = 100;
  const interval = 1;

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(
      <CounterApp value={initialValue} interval={interval} />
    );
    expect(container).toMatchSnapshot();
  });
  test('Debe mostar el valor innicial de "100"', () => {
    render(<CounterApp value={initialValue} interval={interval} />);
    expect(screen.getByText('100')).toBeTruthy();
  });
  test('Debe incrementar con el boton "+1"',()=>{
    render(<CounterApp value={initialValue} interval={interval} />);
    fireEvent.click(screen.getByRole('button', {name:'button-increment'}))
    fireEvent.click(screen.getByRole('button', {name:'button-increment'}))
    expect(screen.getByRole('heading', {level:2}).innerHTML).toBe("102")
  })
  test('Debe decrementar con el boton "-1"',()=>{
    render(<CounterApp value={initialValue} interval={interval} />);
    fireEvent.click(screen.getByRole('button', {name:'button-decrement'}))
    fireEvent.click(screen.getByRole('button', {name:'button-decrement'}))
    expect(screen.getByRole('heading', {level:2}).innerHTML).toBe("98")
  })
  test('Debe reiniciar con el boton "Reset"',()=>{
    render(<CounterApp value={initialValue} interval={interval} />);
    fireEvent.click(screen.getByRole('button', {name:'button-decrement'}))
    fireEvent.click(screen.getByRole('button', {name:'button-decrement'}))
    fireEvent.click(screen.getByRole('button', {name:'button-decrement'}))
    fireEvent.click(screen.getByRole('button', {name:'button-decrement'}))
    expect(screen.getByRole('heading', {level:2}).innerHTML).toBe("96")
    fireEvent.click(screen.getByRole('button', {name:'button-reset'}))
    expect(screen.getByRole('heading', {level:2}).innerHTML).toBe("100")
  })
});
