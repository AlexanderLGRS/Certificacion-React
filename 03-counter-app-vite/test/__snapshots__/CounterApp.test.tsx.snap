// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en App Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button
    aria-label="button-increment"
  >
     + 
  </button>
  <button
    aria-label="button-decrement"
  >
     - 
  </button>
  <button
    aria-label="button-reset"
  >
     Reset 
  </button>
</div>
`;
